# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1709828331 -3600
# Parent  84ef9cba660502f68e8a1da54f0a9af40d85c153
Feature: Make opening <mid: ...> links open faster.

diff --git a/mail/modules/MailUtils.jsm b/mail/modules/MailUtils.jsm
--- a/mail/modules/MailUtils.jsm
+++ b/mail/modules/MailUtils.jsm
@@ -393,9 +393,16 @@ var MailUtils = {
    *
    * @param {string} msgId - The message id string without the brackets.
    */
   openMessageByMessageId(msgId) {
-    let msgHdr = this.getMsgHdrForMsgId(msgId);
+    // The following call will search all folders which is slow and blocks
+    // the UI. Let's give it the current server as start server.
+    let msgHdr = this.getMsgHdrForMsgId(
+      msgId,
+      Services.wm
+        .getMostRecentWindow("mail:3pane")
+        .document.getElementById("tabmail")?.currentAbout3Pane?.gFolder?.server
+    );
     if (msgHdr) {
       this.displayMessage(msgHdr);
       return;
     }
